www/_include.php

```diff
@@ -96,13 +96,23 @@ function __toString() {
 /* Set the $SIMPLESAML_INCPREFIX to a reference to the class. */
 $SIMPLESAML_INCPREFIX = new SimpleSAML_IncPrefixWarn();
 
+/* Get environment variable that specifies an alernate location for the configuration files. */
+$configdir = getenv("SIMPLESAMLPHP_CONFIG_DIR");
 
-$configdir = dirname(dirname(__FILE__)) . '/config';
-if (!file_exists($configdir . '/config.php')) {
-	header('Content-Type: text/plain');
-	echo("You have not yet created the simpleSAMLphp configuration files.\n");
-	echo("See: http://rnd.feide.no/content/installing-simplesamlphp#id434777\n");
-	exit(1);
+/* Check if environment variable is set and if the directory exists.
+ * If neither condition is met use the default configuration directory.
+ */
+if (empty($configdir) || !file_exists($configdir)) {
+	$configdir = dirname(dirname(__FILE__)) . '/config';
+	if (!file_exists($configdir . '/config.php')) {
+		header('Content-Type: text/plain');
+		echo("You have not yet created the simpleSAMLphp configuration files.\n");
+		echo("See: http://rnd.feide.no/content/installing-simplesamlphp#id434777\n");
+		exit(1);
+	}
+} else {
```

Comment:
Since you used `file_exists()`, here you are basically having the following condition:

```php
if (!empty($configdir) && file_exists($configdir)) {
```

Note that in that case, if _SIMPLESAMLPHP_CONFIG_DIR_ is set, for instance, to _/etc/passwd_, you would end up here, causing an error since that's not a directory. `is_dir()` should be used to make sure the path specified is a directory.